### PR TITLE
DP-18678 Add special announcement metadata

### PR DESCRIPTION
**Pull request by tom-fleming**
**Description:**
This work ultimately adds the [SpecialAnnouncement](https://schema.org/SpecialAnnouncement) metadata option. It is adapted from this drupal.org patch: https://www.drupal.org/files/issues/2020-04-22/3120412-12-specialAnnouncement-support.patch. Several modifications have been made to accommodate mass.gov's testing suite (Id), to include tags that were not in the original patch but are recommended by [Google](https://developers.google.com/search/docs/data-types/special-announcements#data-type-definitions) (GovernmentBenefitsInfo and QuarantineGuidelines), and to add options that are needed by this project (making State available in SpatialCoverage). I have also excluded one metatag schema from the patch due to several issues getting it to work properly (SchemaUrlWebContentBase). Lastly, tests are added to LocationSpecialAnnouncementMetadataTest, which is only testing against the Location content type at the moment.

**Jira:** (Skip unless you are MA staff)
https://jira.mass.gov/browse/DP-18678

**To Test:**
- [ ] Go to Configuration > Search and metadata > Metatag > Settings (/admin/config/search/metatag/settings)
- [ ] Verify that SpecialAnnouncement is added to info details, binder, news, location, curated list, executive order, advisory, decision, regulation, promotional page, rules of court, service page
- [ ] Edit content of one or multiple of the above content types. Verify that you can see the SpecialAnnouncement option and its fields when expanded.
- [ ] Enter SpecialAnnouncement metadata content and save the page.
- [ ] Use your browser's inspector and check the contents of the application/ld+json <script> to verify that it includes the SpecialAnnouncement metadata that you entered.
- [ ] Verify ld+json output with Google [here](https://search.google.com/test/rich-results). A template displaying the minumum required/recommended fields can be found [here](https://jira.mass.gov/browse/DP-18678?focusedCommentId=325785&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-325785)

**Screenshots/GIFs:**
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

File: changelogs/DP-18678.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# You may add more than 1 description & issue for each type using the following format:
+# Changed:
+#  - description: Automating the release branch.
+#    issue: DP-10166
+#  - description: Second change item that needs a description.
+#    issue: DP-19875
+#  - description: Third change item that needs a description along with an issue.
+#    issue: DP-19843
+#
+Type:
```
Comment (weitzman) [inline changelogs/DP-18678.yml]: The word Type should be replaced with 'Changed, Added, Fixed etc. See text above.

Comment (tom-fleming) [reply changelogs/DP-18678.yml]: Thanks! This should be resolved.

**Comment by weitzman** [comment]
We typiically add testing for Metatags. See classes that extend \Drupal\Tests\mass_metatag\ExistingSite\MetadataTestCase

**Review by weitzman — APPROVED** [review]
Code looks solid. Anyone is welcome to merge this once it has passed External review

**Pull request merged**